tests/test_mvn_entropy.py

```diff
@@ -0,0 +1,240 @@
+    log_w_sir_upper_all = vmap(
+        log_sir_weight_upper, in_axes=(0, 0, None, None, None, None)
+    )(
+        upper_sir_keys,
+        y_samples_for_upper,
+        model_args,
+        q0_args,
+        P_sir_particles,
+        target_idx_to_estimate,
+    )
+    hat_H_Zi = -jnp.mean(log_w_sir_upper_all)
+
+    # Lower bound H_tilde(Z_i)
+    joint_sampling_keys = random.split(key_joint_sampling, n_outer_samples)
+    joint_traces = vmap(joint_model.simulate, in_axes=(0, None))(
+        joint_sampling_keys, model_args
+    )
+    joint_choices_for_lower = vmap(lambda tr: tr.get_choices())(joint_traces)
+
+    lower_sir_keys = random.split(key_lower_sir, n_outer_samples)
+    log_w_sir_lower_all = vmap(
+        log_sir_weight_lower, in_axes=(0, 0, None, None, None, None)
+    )(
+        lower_sir_keys,
+        joint_choices_for_lower,
+        model_args,
+        q0_args,
+        P_sir_particles,
+        target_idx_to_estimate,
+    )
+    tilde_H_Zi = -jnp.mean(log_w_sir_lower_all)
+
+    return analytical_entropy_i, tilde_H_Zi, hat_H_Zi
+
+
+# JIT compile the main estimation function
+estimate_single_marginal_entropy_bounds_jit = jit(
+    estimate_single_marginal_entropy_bounds,
+    static_argnames=("n_outer_samples", "P_sir_particles", "target_idx_to_estimate"),
+)
+
+
+def test_mvn_entropy():
+    """Runs a simplified demo for estimating a single marginal MVN entropy."""
+    key = random.PRNGKey(42)  # Fixed seed for reproducibility
+
+    # --- Demo Configuration ---
+    dim_z = 2
+    target_idx_to_estimate = 0  # Estimate H(Z_0)
+    n_outer_samples = 1000  # Number of samples for E[...] (outer loop)
+    P_sir_particles = 100  # Number of particles for SIR (inner loop)
+
+    # Define simple MVN parameters
+    mu_full = jnp.array([0.5, -0.5], dtype=jnp.float32)
+    cov_full = jnp.array(
+        [[1.0, 0.7], [0.7, 1.5]], dtype=jnp.float32  # Z0 variance = 1.0
+    )  # Z1 variance = 1.5
+
+    print("--- Simplified MVN Marginal Entropy Estimation Demo ---")
+    print(f"Estimating H(Z_{target_idx_to_estimate}) for a {dim_z}-D MVN:")
+    print(f"  mu_full = {mu_full}")
+    print(f"  cov_full = \n{cov_full}")
+    print(f"  n_outer_samples = {n_outer_samples}")
+    print(f"  P_sir_particles = {P_sir_particles}\n")
+
+    # Run the estimation
+    analytical_H, lower_bound_H, upper_bound_H = (
+        estimate_single_marginal_entropy_bounds_jit(
+            key,
+            n_outer_samples,
+            P_sir_particles,
+            mu_full,
+            cov_full,
+            target_idx_to_estimate,
+        )
+    )
+
+    print(f"Results for H(Z_{target_idx_to_estimate}):")
+    print(f"  Analytical Entropy          : {analytical_H:.4f}")
+    print(f"  Estimated Lower Bound (SIR) : {lower_bound_H:.4f}")
+    print(f"  Estimated Upper Bound (SIR) : {upper_bound_H:.4f}")
+    print(f"  Width of Bounds             : {upper_bound_H - lower_bound_H:.4f}")
```

Comment:
The estimated upper bound (1.4158) is less that the analytical entropy (1.4189). Is there s bug? 🐛 
```bash
--- Simplified MVN Marginal Entropy Estimation Demo ---
Estimating H(Z_0) for a 2-D MVN:
  mu_full = [ 0.5 -0.5]
  cov_full = 
[[1.  0.7]
 [0.7 1.5]]
  n_outer_samples = 1000
  P_sir_particles = 100

Results for H(Z_0):
  Analytical Entropy          : 1.4189
  Estimated Lower Bound (SIR) : 1.3912
  Estimated Upper Bound (SIR) : 1.4158
  Width of Bounds             : 0.0246
```